src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/ReplicationCheckerTest.java

```diff
@@ -0,0 +1,245 @@
+        List<?> notReplicated = (List<?>) resourceMap.get("notReplicatedTaskIds");
+        assertEquals(0, notReplicated.size());
+        verify(cftTaskDatabaseService, times(1)).findLastFiveUpdatedTasks();
+    }
+
+    @Test
+    void should_process_replication_checker_operation_for_found_history() {
+        TaskResource resource = new TaskResource("1",
+            OffsetDateTime.parse("2021-05-09T20:15:50.345875+01:00"), "claim", "someuser");
+        TaskHistoryResource history = new TaskHistoryResource("1",
+            OffsetDateTime.parse("2021-05-09T20:15:50.345875+01:00"), "claim", "someuser");
+        when(cftTaskDatabaseService.findLastFiveUpdatedTasks()).thenReturn(List.of(resource));
+        when(miReportingService.findByTaskIdOrderByLatestUpdate("1")).thenReturn(List.of(history));
+
+        Map<String, Object> resourceMap = replicationChecker.performOperation(request).getResponseMap();
+        List<?> tasks = (List<?>) resourceMap.get("replicationCheckedTaskIds");
+        assertEquals(1, tasks.size());
+        List<?> notReplicated = (List<?>) resourceMap.get("notReplicatedTaskIds");
+        assertEquals(0, notReplicated.size());
+
+        verify(cftTaskDatabaseService, times(1)).findLastFiveUpdatedTasks();
+
+        await()
+            .pollInterval(1, SECONDS)
+            .atMost(20, SECONDS)
+            .until(() -> {
+                verify(miReportingService, times(1)).findByTaskIdOrderByLatestUpdate("1");
+                return true;
+            });
+
+    }
+
+    @Test
+    void should_process_replication_checker_operation_for_not_found_history_timestamp_mismatch() {
+        TaskResource resource = new TaskResource("1",
+            OffsetDateTime.parse("2021-05-09T20:15:50.345875+01:00"), "claim", "someuser");
+        TaskHistoryResource history = new TaskHistoryResource("1",
+            OffsetDateTime.parse("2021-05-09T20:15:30.345875+01:00"), "claim", "someuser");
+        when(cftTaskDatabaseService.findLastFiveUpdatedTasks()).thenReturn(List.of(resource));
+        when(miReportingService.findByTaskIdOrderByLatestUpdate("1")).thenReturn(List.of(history));
+
+        Map<String, Object> resourceMap = replicationChecker.performOperation(request).getResponseMap();
+        List<?> tasks = (List<?>) resourceMap.get("replicationCheckedTaskIds");
+        assertEquals(1, tasks.size());
+        List<?> notReplicated = (List<?>) resourceMap.get("notReplicatedTaskIds");
+        assertEquals(1, notReplicated.size());
+
+        verify(cftTaskDatabaseService, times(1)).findLastFiveUpdatedTasks();
+
+        await()
+            .pollInterval(1, SECONDS)
+            .atMost(20, SECONDS)
+            .until(() -> {
+                verify(miReportingService, atLeast(2)).findByTaskIdOrderByLatestUpdate("1");
+                return true;
+            });
+
+    }
+
+    @Test
+    void should_process_replication_checker_operation_for_not_found_history_action_mismatch() {
```

Comment:
We can use CapturedOutput to verify the replicationLog

Reply:
Added log assertion for all error scenario tests